From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Thu, 25 Jul 2019 19:33:53 +0200
Subject: [PATCH] {PRINTF} drivers/misc/habanalabs/goya: use %pad to format
 dma_addr_t
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Compiling with gcc-x86_32 leads to:

    In file included from include/linux/printk.h:332,
                     from include/linux/kernel.h:15,
                     from include/linux/list.h:9,
                     from include/linux/kobject.h:19,
                     from include/linux/cdev.h:5,
                     from drivers/misc/habanalabs/habanalabs.h:14,
                     from drivers/misc/habanalabs/goya/goyaP.h:12,
                     from drivers/misc/habanalabs/goya/goya.c:8:
    drivers/misc/habanalabs/goya/goya.c: In function ‘goya_sw_init’:
    drivers/misc/habanalabs/goya/goya.c:698:21: error: format ‘%llx’
    expects argument of type ‘long long unsigned int’, but argument 4
    has type ‘dma_addr_t’ {aka ‘unsigned int’} [-Werror=format=]
      698 |  dev_dbg(hdev->dev, "cpu accessible memory at bus address 0x%llx\n",
          |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    include/linux/dynamic_debug.h:125:15: note: in definition of macro ‘__dynamic_func_call’
      125 |   func(&id, ##__VA_ARGS__);  \
          |               ^~~~~~~~~~~
    include/linux/dynamic_debug.h:157:2: note: in expansion of macro ‘_dynamic_func_call’
      157 |  _dynamic_func_call(fmt,__dynamic_dev_dbg,   \
          |  ^~~~~~~~~~~~~~~~~~
    include/linux/device.h:1503:2: note: in expansion of macro ‘dynamic_dev_dbg’
     1503 |  dynamic_dev_dbg(dev, dev_fmt(fmt), ##__VA_ARGS__)
          |  ^~~~~~~~~~~~~~~
    include/linux/device.h:1503:23: note: in expansion of macro ‘dev_fmt’
     1503 |  dynamic_dev_dbg(dev, dev_fmt(fmt), ##__VA_ARGS__)
          |                       ^~~~~~~
    drivers/misc/habanalabs/goya/goya.c:698:2: note: in expansion of macro ‘dev_dbg’
      698 |  dev_dbg(hdev->dev, "cpu accessible memory at bus address 0x%llx\n",
          |  ^~~~~~~
    drivers/misc/habanalabs/goya/goya.c:698:64: note: format string is defined here
      698 |  dev_dbg(hdev->dev, "cpu accessible memory at bus address 0x%llx\n",
          |                                                             ~~~^
          |                                                                |
          |                                                                long long unsigned int
          |                                                             %x
---
 drivers/misc/habanalabs/goya/goya.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/misc/habanalabs/goya/goya.c b/drivers/misc/habanalabs/goya/goya.c
index 5a95ff4489ad..4c4b7de20c4f 100644
--- a/drivers/misc/habanalabs/goya/goya.c
+++ b/drivers/misc/habanalabs/goya/goya.c
@@ -695,8 +695,8 @@ static int goya_sw_init(struct hl_device *hdev)
 		goto free_dma_pool;
 	}
 
-	dev_dbg(hdev->dev, "cpu accessible memory at bus address 0x%llx\n",
-		hdev->cpu_accessible_dma_address);
+	dev_dbg(hdev->dev, "cpu accessible memory at bus address %pad\n",
+		&hdev->cpu_accessible_dma_address);
 
 	hdev->cpu_accessible_dma_pool = gen_pool_create(ilog2(32), -1);
 	if (!hdev->cpu_accessible_dma_pool) {
-- 
